quint/src/builtin.qnt

```diff
@@ -835,6 +835,24 @@ module builtin {
   /// ```
   action assert(p): (bool) => bool
 
+  /// `debug(msg, value)` prints the given message and value to the console,
+  /// separated by a space.
+  ///
+  /// It also returns the given value unchanged,
+  /// so that it can be used directly within expressions.
+  ///
+  /// ### Examples
+  ///
+  /// ```
+  /// var x: int
+  /// >>> (x' = 0).then(3.reps(i => x' = debug("new x:", x + 1)))
+  /// > new x: 1
+  /// > new x: 2
+  /// > new x: 3
+  /// true
+  /// ```
+  pure def debug(msg, value): (str, a) => a
```

Comment:
Can a pure function read from state variables? If not, then we need to remove `pure` here.

Reply:
a pure function cannot read the state variables. So yeah, it should not be pure.

Reply:
Actually, pure defs can operate on values that read from the state, but it can't add more reads on top of that.

`f` is pure here:
```
var s: int
pure f(x) = x

val foo = f(s)
```

`f` is NOT pure here:
```
var s: int
pure f(x) = x + s

val foo = f(1)
```

So in this case, `debug` can be pure

Reply:
Ah, I have missed that. You are right @bugarela